### DISCUSSION TITLE
Error message when running a preview

### DISCUSSION THREAD

**Discussion by zsofiaj** [thegetty/quire#164] [Get Help] [answered]
Hi everyone! I've just installed Quire on a new Mac, and I can't get the preview to run. (Everything is still working fine on my old laptop.) When I run "quire preview" or "quire preview --verbose," I see the error message I've pasted below. Has anyone encountered this, and any idea how to fix it? I see someone named Darcy had this problem back in April, but I wasn't able to find a solution in that thread. 

Thank you!

Launching preview server
ℹ Navigate to http://localhost:1313 to see your changes.
ℹ Press Control+C or type "quire stop" to stop the preview.
events.js:377
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules/.bin/webpack ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:467:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
    at onErrorNT (internal/child_process.js:467:16)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node_modules/.bin/webpack',
  path: 'node_modules/.bin/webpack',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack/webpack.config.dev.js'
  ]
}

<img width="728" alt="Screen Shot 2021-08-25 at 8 35 26 AM" src="https://user-images.githubusercontent.com/68071544/130791968-fd133587-aca6-4e83-94c8-dcf47716430d.png">

**Comment by Erin-Cecele** [comment, marked as answer]
Hi @zsofiaj did you run `quire install` prior to running `quire preview`? This is a required step the first time you work in a newly cloned repository. Let us know if that fixes things.

**Reply by zsofiaj** [reply]
That worked! Thanks so much, @Erin-Cecele. I hadn't cloned a repository in a while, and I completely forgot about that step. Next time if I get stuck, I'll make sure to refer back to the documentation on cloning a repository. 

One quick follow-up question: 

When I ran "quire install," it generated a warning at the bottom, see below. Should I update npm, as suggested? The preview does seem to be working fine now.

<img width="843" alt="Screen Shot 2021-08-27 at 1 24 10 PM" src="https://user-images.githubusercontent.com/68071544/131165736-dbedda9d-05a1-450f-b2be-440564e50a2a.png">

**Reply by Erin-Cecele** [reply]
@zsofiaj Wonderful. I'm glad that worked for you. You can ignore the npm message. With Quire, you only really need to worry about any messages that say ERROR or ERR. Don't hesitate to reach out if you have other questions!

**Reply by zsofiaj** [reply]
@Erin-Cecele Will ignore that npm message. Thanks so much!

**Reply by robertweisbergmetmuseum** [reply]
Hi, @Erin-Cecele I hope it's okay that I'm resurrecting this question, but I'm getting a similar set of error messages when I try running a cloned repository on my new computer. It's slightly different: 
Launching preview server
ℹ Navigate to http://localhost:1313 to see your changes.
ℹ Press Control+C or type "quire stop" to stop the preview.
node:events:504
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules/.bin/webpack ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node_modules/.bin/webpack',
  path: 'node_modules/.bin/webpack',
  spawnargs: [
    '--watch',
    '-d',
    '--progress',
    '--color',
    '--config',
    'webpack/webpack.config.dev.js'
  ]
}

I have another repository in the same folder on my computer. In my user folder I have a folder called GitHub_projects, and now in this folder I have two project folders. The other project is previewing successfully but running quire install didn't prevent the error messages when I try previewing the new project. (I did use cd to move to the other project's folder.) Any idea what I still might be doing wrong? (I am quite rusty on Quire, it's been a while since I've tried it out.)
Rob

**Comment by Erin-Cecele** [comment]
Hi @robertweisbergmetmuseum would it be possible to attach the full text from Terminal? You can export it by going to Shell / Export Text As. That will help me follow along with your navigation. Thank you!

**Comment by robertweisbergmetmuseum** [comment]
Hi, Erin,

Attached the the terminal output.

I'm moving on to another project for the rest of the afternoon so there's no rush to reply. Thanks!

R
________________________________
From: Erin Cecele Dunigan ***@***.***>
Sent: Thursday, April 14, 2022 3:52 PM
To: thegetty/quire ***@***.***>
Cc: Weisberg, Robert ***@***.***>; Mention ***@***.***>
Subject: [External] - Re: [thegetty/quire] Error message when running a preview (Discussion #164)


Hi @robertweisbergmetmuseum<https://github.com/robertweisbergmetmuseum> would it be possible to attach the full text from Terminal? You can export it by going to Shell / Export Text As. That will help me follow along with your navigation. Thank you!

**Reply by Erin-Cecele** [reply]
Hi Robert, I don't see an attachment. If you go into GitHub Discussions in your web browser, you can drag-and-drop it into the response field: https://github.com/thegetty/quire/discussions/164#discussioncomment-2569425. But no rush!

**Comment by Erin-Cecele** [comment]
Hi @robertweisbergmetmuseum are you on the macOS Monterey? If yes, please see discussion #283 regarding reinstalling Python. Let me know if that is helpful!
